torch/_ops.py

```diff
@@ -246,6 +246,10 @@ class HigherOrderOperator(OperatorBase):
     # practice due to name collisions.
     def __init__(self, name):
         super().__init__()
+        if type(self) is HigherOrderOperator:
+            raise RuntimeError(
+                "Direct instantiation of HigherOrderOperator is not allowed. Please subclass it."
+            )
```

Comment:
Next step: enforce that all HOPs override `__call__` so that they actually have a signature and a docstring. (feel free to file a follow-up issue for this if you don't want to handle it right now)

Reply:
Track it https://github.com/pytorch/pytorch/issues/133719